homeassistant/components/hydrawise/manifest.json

```diff
@@ -1,9 +1,11 @@
 {
   "domain": "hydrawise",
   "name": "Hunter Hydrawise",
-  "codeowners": ["@ptcryan"],
+
+  "codeowners": ["@dknowles", "@ptcryan"],
+  "config_flow": true,
   "documentation": "https://www.home-assistant.io/integrations/hydrawise",
   "iot_class": "cloud_polling",
   "loggers": ["hydrawiser"],
-  "requirements": ["hydrawiser==0.2"]
+  "requirements": ["hydrawiser==0.2", "pydrawise==2023.1.4"]
```

Comment:
The hydrawiser library is no longer maintained. (See: https://github.com/ptcryan/hydrawiser which has had no activity since 2020, including responses to bugs or PRs)

There's also a newer API from Hydrawise that's implemented using GraphQL which is much more robust both in terms of features, reliability, and supportability. (it's what their website and mobile app use)

I don't think there's a clean way to do a drop-in replacement, for a few reasons:

1. The authentication mechanism changes with the new API
2. The old API returns truncated strings for things, which will cause problems in a migration
3. I don't think there are stable identifiers used across both APIs, so we couldn't make unique IDs in HA and then migrate libraries

Since it's already a breaking change to the user to migrate from YAML to config flow, this seems like a reasonable time to also change the underlying API.

Reply:
I've looked at this again.
I think you should postpone the new library until after config flow has been implemented.

Reply:
Okay, sounds good. I sent another PR (https://github.com/home-assistant/core/pull/92305) to kick this off by adding a `DataUpdateCoordinator` to the code. It should make adding config flow simpler.